### PR TITLE
Merge pull request #2 from GrindLabsOrg/main

### PR DESCRIPTION
**Pull request by uri-99**
merge from setup to main

### CODE REVIEW

File: src/yas_meme.cairo [hunk 1]
```diff
@@ -1,22 +1,90 @@
+use starknet::ContractAddress;
+// use bytes_31::split_felt252;
```
Comment (rcatalan98) [inline src/yas_meme.cairo:2]:
```suggestion
```

File: src/yas_meme.cairo [hunk 2]
```diff
@@ -1,22 +1,90 @@
...
+            assert(get_caller_address() == self.owner.read(), 'You are not the owner');
+            self.title.write(new_title);
+            true
+        }
+
+        fn read_caption(self: @ContractState) -> Array<felt252>{
+            self.caption.read()
+        }
+
+        fn write_caption(ref self: ContractState, new_caption: Array<felt252>) -> bool{
```
Comment (rcatalan98) [inline src/yas_meme.cairo:80]: Why are we returning a `bool`?

**Review by rcatalan98 — CHANGES_REQUESTED** [review]

File: src/yas_meme.cairo (hunk 2 above)
Comment (uri-99) [reply src/yas_meme.cairo:80]: because i made a mistake, thanks for pointing it out !

**Review by rcatalan98 — APPROVED** [review]
Great job!

**Pull request merged**